spec/typed_parameters/comparison_spec.rb

```diff
@@ -25,10 +25,25 @@
       end
 
       context "when the request body does not adhere to the rules format" do
-        let(:name) { 1_000 }
+        context "when a non-specified parameter is included in the request body" do
+          let(:params) { { name: "M@", email: "matt@zipmark.com" } }
 
-        it "returns an object describing the invalid attribute" do
-          expect(subject).to eq({ "name" => "param_must_be_string" })
+          # NOTE: pick one of the following 2 test cases as desired behavior
```

Comment:
@kurko do you think 'extra' params should simply be ignored, or should we actually mark the request as being invalid (this is what we're doing now).

Example:

```ruby
Rules = { data: Hash }

# the `foo` param is not whitelisted
param = { data: {}, foo: "bar" }
```